modules/core/04-channel/keeper/ante.go

```diff
@@ -0,0 +1,22 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+)
+
+// RecvPacketReCheckTx applies replay protection ensuring that when relay messages are
+// re-executed in ReCheckTx, we can appropriately filter out redundant relay transactions.
+func (k *Keeper) RecvPacketReCheckTx(ctx sdk.Context, packet types.Packet) error {
```

Comment:
Wondering if there would be an alternative way to implement this where we don't need to increase the public API of the channel keeper...

Also, for my understanding: why what we do in `ReCheckTx` is different from what we do in `CheckTx`?

Reply:
ReCheckTx is rerun on the mempool after every block commit. It is redundant to recheck proofs and other logic which will not change between commits